workers/unity/Packages/io.improbable.gdk.core/Dynamic/ComponentDatabase.cs

```diff
@@ -21,6 +25,12 @@ static ComponentDatabase()
 
             ComponentsToIds = Metaclasses.ToDictionary(pair => pair.Value.Data, pair => pair.Key);
             SnapshotsToIds = Metaclasses.ToDictionary(pair => pair.Value.Snapshot, pair => pair.Key);
+            RequestsToCommandMetaclass = Metaclasses.Where(pair => pair.Value.Commands.Length > 0)
+                .SelectMany(pair => pair.Value.Commands.Select(cmd => (cmd.Request, cmd)))
+                .ToDictionary(pair => pair.Request, pair => pair.cmd);
+            ReceivedRequestsToCommandMetaclass = Metaclasses.Where(pair => pair.Value.Commands.Length > 0)
+                .SelectMany(pair => pair.Value.Commands.Select(cmd => (cmd.ReceivedRequest, cmd)))
+                .ToDictionary(pair => pair.ReceivedRequest, pair => pair.cmd);
```

Comment:
```suggestion
            RequestsToCommandMetaclass = Metaclasses
                .SelectMany(pair => pair.Value.Commands.Select(cmd => (cmd.Request, cmd)))
                .ToDictionary(pair => pair.Request, pair => pair.cmd);
            ReceivedRequestsToCommandMetaclass = Metaclasses
                .SelectMany(pair => pair.Value.Commands.Select(cmd => (cmd.ReceivedRequest, cmd)))
                .ToDictionary(pair => pair.ReceivedRequest, pair => pair.cmd);
```

The `Where` is redundant as the inner closure of `SelectMany` will return an empty enumerable for the case where `Commands.Length == 0`